sockeye/lr_scheduler.py

```diff
@@ -34,21 +55,23 @@ class LearningRateSchedulerInvSqrtT(LearningRateScheduler):
 
     :param updates_per_checkpoint: Number of batches between checkpoints.
     :param half_life: Half life of the learning rate in number of checkpoints.
+    :param warmup: Number of (linear) learning rate increases to warm-up.
     """
 
-    def __init__(self, updates_per_checkpoint: int, half_life: int) -> None:
+    def __init__(self, updates_per_checkpoint: int, half_life: int, warmup: int = 0) -> None:
+        super().__init__(warmup)
         check_condition(updates_per_checkpoint > 0, "updates_per_checkpoint needs to be > 0.")
         check_condition(half_life > 0, "half_life needs to be > 0.")
-        # Note: will be overwritten by optimizer  in mxnet
-        self.base_lr = None
         # 0.5 base_lr = base_lr * sqrt(1 + T * factor)
         # then factor = 3 ./ T, with T = half_life * updates_per_checkpoint
         self.factor = 3. / (half_life * updates_per_checkpoint)
+        self.warmup = warmup
         self.t_last_log = -1
         self.log_every_t = int(half_life * updates_per_checkpoint)
 
     def __call__(self, num_updates: int):
-        lr = self.base_lr / sqrt(1 + num_updates * self.factor)
+        change = min(1.0 / sqrt(1 + num_updates * self.factor), self._warmup(num_updates) if self.warmup > 0 else 99999)
+        lr = self.base_lr * change
```

Comment:
Somehow can't see my other comment, but warmup should either return a fraction (instead of a learning rate) or the result here shouldn't be multiplied by base_lr. Otherwise you multiply by base_lr twice.